modules/internal/ChapelArray.chpl

```diff
@@ -309,33 +305,74 @@ module ChapelArray {
                       " If so, use {...} instead of [...].");
     }
 
+    param homog = isHomogeneousTuple(elems);
+
     // elements of string literals are assumed to be of type string
-    type eltType = _getLiteralType(elems(0).type);
+    type eltType = if homog then elems(0).type
+                            else chpl_computeUnifiedType(elems);
```

Comment:
Perhaps computeUnifiedType() does not need to cater to the homogeneous case?

Reply:
It definitely shouldn't have to in the current code organization.  I made it handle it anyway just for generality (rather than adding a where clause only making the routine apply to het. tuples, say), and in case it saw additional uses in the future (though I hope dyno will replace this functionality before that happens).